### PR TITLE
Fix for supervisor access

### PR DESCRIPTION
**Pull request by stepanov**

### CODE REVIEW

File: src/Event/Model/ModelBeforeFindEventsListener.php [hunk 1]
```diff
@@ -102,16 +102,23 @@ public function filterByUserCapabilities(Event $event, Query $query, ArrayObject
             return;
         }
 
+        $result = 0;
+        $result |= $this->filterQuery($query, $table, $user, $controllerName);
```
Comment (mamchenkov) [inline src/Event/Model/ModelBeforeFindEventsListener.php]: @stepanov I haven't seen this operator in PHP.  Can you find a reference somewhere? Just to check if it works on both PHP 5 and PHP 7.

Comment (stepanov) [reply src/Event/Model/ModelBeforeFindEventsListener.php]: @mamchenkov http://php.net/manual/en/language.operators.bitwise.php - it's the same as $result = $result | exp();

Comment (mamchenkov) [reply src/Event/Model/ModelBeforeFindEventsListener.php]: @stepanov I don't think it works.  These bitwise operators, which work with bits and integers.  Theresult of `filterQuery()` is an object, if I remember correctly.

Here is the test I did:

```php
<?php

$class = new StdClass();
$result = 0;
$result |= $class;

print_r($result);
```

The result:

```
$ php z.php 
PHP Notice:  Object of class stdClass could not be converted to int in /home/leonid/z.php on line 5
PHP Stack trace:
PHP   1. {main}() /home/leonid/z.php:0

Notice: Object of class stdClass could not be converted to int in /home/leonid/z.php on line 5

Call Stack:
    0.0001     397080   1. {main}() /home/leonid/z.php:0

1
```

Line 5 is the one that does `|=`.

Comment (mamchenkov) [reply src/Event/Model/ModelBeforeFindEventsListener.php]: @stepanov my bad ... then we can just set the `$result=$this->filterQuery(...)` ...

File: src/Event/Model/ModelBeforeFindEventsListener.php [hunk 2]
```diff
@@ -128,9 +135,10 @@ public function filterByUserCapabilities(Event $event, Query $query, ArrayObject
      * @param \Cake\ORM\Table $table Table instance
      * @param array $user User info
      * @param string $controllerName Namespaced controller name
+     * @param bool $useOr specified how to connect where statements - via AND (default) or OR
      * @return void
```
Comment (mamchenkov) [inline src/Event/Model/ModelBeforeFindEventsListener.php]: @stepanov can you please update this docblock for the return value type and description.  Thanks.

**Pull request merged**